File: FrontEnd/src/app/models/book.dto.ts
export class BookDTO {
    id!: number;
    title!: string;
    author!: string;
    description!: string;
    state!: boolean;
    nbPage!:number;
    imageUrl!: string;
    price!: number;
    categoryId!: number;
    isReturned!:boolean;
    isApproved!:boolean;
    requestReturn!: boolean;
    endDate!:string;
    remainingDays?: number;
    recommended!:boolean;
}

File: BackEnd/src/main/java/com/example/futurbe/dto/BookDTO.java
package com.example.futurbe.dto;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
public class BookDTO {
    private long id;
    private String title;
    private String author;
    private String description;
    private Boolean state;
    private String imageUrl;
    private float price;
    private Long categoryId;
    private int nbPage;
    private boolean recommended;
    private Boolean isReturned;
    private Boolean isApproved;
    private Boolean requestReturn;
    private LocalDate endDate;
}
